From 934cf0516d8824e26321119198098267cd7539b4 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 22 Dec 2018 00:19:58 +0100
Subject: [PATCH 1/1] efi_selftest: block device: avoid read after free

Reading the position in a file after closing the same results in a read
after free.

Correct the sequence in the test.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_selftest/efi_selftest_block_device.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/efi_selftest/efi_selftest_block_device.c b/lib/efi_selftest/efi_selftest_block_device.c
index d4e4fac1c74..f038da9f190 100644
--- a/lib/efi_selftest/efi_selftest_block_device.c
+++ b/lib/efi_selftest/efi_selftest_block_device.c
@@ -445,11 +445,6 @@ static int execute(void)
 		efi_st_error("Failed to write file\n");
 		return EFI_ST_FAILURE;
 	}
-	ret = file->close(file);
-	if (ret != EFI_SUCCESS) {
-		efi_st_error("Failed to close file\n");
-		return EFI_ST_FAILURE;
-	}
 	ret = file->getpos(file, &pos);
 	if (ret != EFI_SUCCESS) {
 		efi_st_error("GetPosition failed\n");
@@ -460,6 +455,11 @@ static int execute(void)
 			     (unsigned int)pos);
 		return EFI_ST_FAILURE;
 	}
+	ret = file->close(file);
+	if (ret != EFI_SUCCESS) {
+		efi_st_error("Failed to close file\n");
+		return EFI_ST_FAILURE;
+	}
 
 	/* Verify file */
 	boottime->set_mem(buf, sizeof(buf), 0);
-- 
2.19.2

